zaza/charm_lifecycle/utils.py

```diff
@@ -63,3 +65,22 @@ def generate_model_name():
     :rtype: str
     """
     return 'zaza-{}'.format(str(uuid.uuid4())[-12:])
+
+
+def check_output_logging(cmd):
+    """Run command and log output.
+
+    :param cmd: Shell command to run
+    :type cmd: List
+    """
+    popen = subprocess.Popen(
+        cmd,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        universal_newlines=True)
+    for line in iter(popen.stdout.readline, ""):
+        logging.info(line.strip())
+    popen.stdout.close()
+    return_code = popen.wait()
```

Comment:
There's a warning in the docs about using `subprocess.PIPE` and `Popen.wait()`: https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait

Is this something we need to code around and use `communicate()` instead?